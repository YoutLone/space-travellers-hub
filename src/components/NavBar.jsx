import React from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import '../styles/NavBar.css';

const NavBar = () => {
  const links = [
    { path: '/', name: 'Rockets' },
    { path: '/missions', name: 'Missions' },
    { path: '/my-profile', name: 'My Profile' },
  ];

  const location = useLocation();

  return (
    <header className="navbar-container">
      <nav className="nav-bar">
        <div className="logo">
          <img src="\resource\planet.png" alt="logo" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
        <ul className="nav-link">
          {links.map((link) => (
            <li key={link.name}>
              <NavLink
                to={link.path}
                className={link.path === location.pathname ? 'active' : ''}
              >
                {link.name}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default NavBar;
