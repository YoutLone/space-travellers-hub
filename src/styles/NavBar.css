.navbar-container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.nav-bar {
  width: 90%;
  padding: 0 1.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 2px solid #e8e8e8;
}

.nav-bar > .logo {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.nav-bar > .logo > h1 {
  font-size: 1.5rem;
  font-weight: 500;
  font-family: Georgia, 'Times New Roman', Times, serif;
}

.nav-bar > .logo > img {
  width: 50px;
  height: 50px;
}

.nav-bar > .nav-link {
  display: flex;
  align-items: center;
  gap: 1.5rem;
}

.nav-bar > .nav-link > li {
  color: #0290ff;
  list-style: none;
}

li :-webkit-any-link {
  text-decoration: none;
  color: #0290ff;
}

li:active :-webkit-any-link {
  text-decoration: underline;
}

li:visited :-webkit-any-link {
  text-decoration: underline;
}

.nav-bar > .nav-link > li:last-child {
  padding-left: 1rem;
  border-left: 2px solid #969595;
}
