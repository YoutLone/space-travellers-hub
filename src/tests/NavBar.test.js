import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import NavBar from '../components/NavBar';

describe('NavBar', () => {
  test('renders navigation links correctly', () => {
    render(
      <Router>
        <NavBar />
      </Router>,
    );
  });

  test('displays the logo and header correctly', () => {
    const { getByAltText, getByText } = render(
      <Router>
        <NavBar />
      </Router>,
    );
    const logo = getByAltText('logo');
    expect(logo).toBeInTheDocument();

    const header = getByText("Space Travelers' Hub");
    expect(header).toBeInTheDocument();
  });

  test('displays the navigation links correctly with correct paths', () => {
    const { getByText } = render(
      <Router>
        <NavBar />
      </Router>,
    );
    const links = [
      { path: '/', text: 'Rockets' },
      { path: '/missions', text: 'Missions' },
      { path: '/my-profile', text: 'My Profile' },
    ];

    links.forEach((link) => {
      const linkElement = getByText(link.text);
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveAttribute('href', link.path);
    });
  });

  test('render NavBar component correctly', () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
