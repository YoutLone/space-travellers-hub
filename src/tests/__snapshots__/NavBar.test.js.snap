// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render NavBar component correctly 1`] = `
<DocumentFragment>
  <header
    class="navbar-container"
  >
    <nav
      class="nav-bar"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="\\\\resource\\\\planet.png"
        />
        <h1>
          Space Travelers' Hub
        </h1>
      </div>
      <ul
        class="nav-link"
      >
        <li>
          <a
            aria-current="page"
            class="active active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/my-profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
